The job **`x86_64-msvc-ext`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8540714716/job/23398381945) [(plain)](https://github.com/rust-lang-ci/rust/commit/62bf84c58a3b2d0327d461351b9b025e968448ef/checks/23398381945/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test result: ok. 24 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.07s

     Running tests\ui.rs (build\x86_64-pc-windows-msvc\stage1-tools\x86_64-pc-windows-msvc\release\deps\ui-5cb20e50bde166fc.exe)
## Running ui tests in tests/pass for x86_64-pc-windows-msvc
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="C:\\a\\_temp\\msys64\\tmp\\miri-uitest-HLQBfE" "RUST_BACKTRACE"="1" C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1\bin\miri.exe "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zmiri-provenance-gc=1" "-Zui-testing" "--target" "x86_64-pc-windows-msvc" "--out-dir" OUT_DIR
tests/pass\align_repeat_into_well_aligned_array.rs ... ok
tests/pass\alloc-access-tracking.rs ... ignored (in-test comment)
tests/pass\align_repeat_into_packed_field.rs ... ok
tests/pass\align_strange_enum_discriminant_offset.rs ... ok
---

test result: ok. 332 passed; 7 ignored;

## Running ui tests in tests/pass-dep for x86_64-pc-windows-msvc
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="C:\\a\\_temp\\msys64\\tmp\\miri-uitest-HLQBfE" "RUST_BACKTRACE"="1" C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1\bin\miri.exe "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zmiri-provenance-gc=1" "-Zui-testing" "--target" "x86_64-pc-windows-msvc" "--out-dir" OUT_DIR
tests/pass-dep\extra_fn_ptr_gc.rs ... ignored (in-test comment)
tests/pass-dep\malloc.rs ... ignored (in-test comment)
tests/pass-dep\tempfile.rs ... ignored (in-test comment)
tests/pass-dep\concurrency\libc_pthread_cond_timedwait.rs ... ignored (in-test comment)
---

test result: ok. 10 passed; 21 ignored;

## Running ui tests in tests/panic for x86_64-pc-windows-msvc
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="C:\\a\\_temp\\msys64\\tmp\\miri-uitest-HLQBfE" "RUST_BACKTRACE"="1" C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1\bin\miri.exe "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zmiri-provenance-gc=1" "-Zui-testing" "--target" "x86_64-pc-windows-msvc" "--out-dir" OUT_DIR
tests/panic\panic3.rs ... ok
tests/panic\overflowing-rsh-2.rs ... ok
tests/panic\panic2.rs ... ok
tests/panic\div-by-zero-2.rs ... ok
---

test result: ok. 12 passed; 1 ignored;

## Running ui tests in tests/fail for x86_64-pc-windows-msvc
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="C:\\a\\_temp\\msys64\\tmp\\miri-uitest-HLQBfE" "RUST_BACKTRACE"="1" C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1\bin\miri.exe "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zmiri-provenance-gc=1" "-Zui-testing" "--target" "x86_64-pc-windows-msvc" "--out-dir" OUT_DIR
tests/fail\deny_lint.rs ... ok
tests/fail\const-ub-checks.rs ... ok
tests/fail\environ-gets-deallocated.rs ... ignored (in-test comment)
tests/fail\coroutine-pinned-moved.rs ... ok
---

test result: ok. 434 passed; 5 ignored;

## Running ui tests in tests/fail-dep for x86_64-pc-windows-msvc
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="C:\\a\\_temp\\msys64\\tmp\\miri-uitest-HLQBfE" "RUST_BACKTRACE"="1" C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1\bin\miri.exe "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zmiri-provenance-gc=1" "-Zui-testing" "--target" "x86_64-pc-windows-msvc" "--out-dir" OUT_DIR
tests/fail-dep\concurrency\libc_pthread_create_main_terminate.rs ... ignored (in-test comment)
tests/fail-dep\concurrency\libc_pthread_create_too_few_args.rs ... ignored (in-test comment)
tests/fail-dep\concurrency\libc_pthread_join_main.rs ... ignored (in-test comment)
tests/fail-dep\concurrency\libc_pthread_create_too_many_args.rs ... ignored (in-test comment)
---
tests/pass-dep\concurrency\windows_init_once.rs ... FAILED
tests/pass-dep\rand.rs ... ok

FAILED TEST: tests/pass-dep\concurrency\windows_init_once.rs
command: "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1\\bin\\miri.exe" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-gnu" "--extern" "getrandom_01=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libgetrandom-8c60e602132deecf.rlib" "--extern" "getrandom_01=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libgetrandom-8c60e602132deecf.rmeta" "--extern" "getrandom_02=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libgetrandom-091aece16d7a4a24.rlib" "--extern" "getrandom_02=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libgetrandom-091aece16d7a4a24.rmeta" "--extern" "libc=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\liblibc-b46113e407866150.rlib" "--extern" "libc=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\liblibc-b46113e407866150.rmeta" "--extern" "num_cpus=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libnum_cpus-2319728e441d867c.rlib" "--extern" "num_cpus=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libnum_cpus-2319728e441d867c.rmeta" "--extern" "rand=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\librand-1f4c3704a3cc87ec.rlib" "--extern" "rand=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\librand-1f4c3704a3cc87ec.rmeta" "--extern" "tempfile=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libtempfile-33242cb722531163.rlib" "--extern" "tempfile=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libtempfile-33242cb722531163.rmeta" "--extern" "page_size=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libpage_size-f337c5a90d549587.rlib" "--extern" "page_size=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libpage_size-f337c5a90d549587.rmeta" "--extern" "tokio=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libtokio-4ce514ae66012c1b.rlib" "--extern" "tokio=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libtokio-4ce514ae66012c1b.rmeta" "--extern" "windows_sys=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libwindows_sys-fcb6ce3fcd33fa4e.rlib" "--extern" "windows_sys=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps\\libwindows_sys-fcb6ce3fcd33fa4e.rmeta" "-L" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\debug\\deps" "-L" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\miri\\i686-pc-windows-gnu\\debug\\deps" "--out-dir" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\ui\\tests\\pass-dep\\concurrency" "tests/pass-dep\\concurrency\\windows_init_once.rs" "-Zmiri-preemption-rate=0" "--edition" "2021"
   0: ui tests in tests/pass-dep for i686-pc-windows-gnu failed
   1: tests failed


Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
error: test failed, to rerun pass `--test ui`
Caused by:
  process didn't exit successfully: `C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1-tools\x86_64-pc-windows-msvc\release\deps\ui-5cb20e50bde166fc.exe pass` (exit code: 1)

##[error]no message
##[error]no message
##[error]you likely need to bless the tests with `./miri test --bless`
##[error]remove this line and possibly later ones by blessing the test
##[error]Error: Undefined Behavior: Data race detected between (1) non-atomic write on thread `unnamed-1` and (2) non-atomic read on thread `unnamed-2` at alloc5320+0x8. (2) just happened here
error: pass test got exit code: 1, but expected 0
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass-dep\concurrency\windows_init_once.stderr` to the actual output
--- tests/pass-dep\concurrency\windows_init_once.stderr
+++ <stderr output>
+++ <stderr output>
+error: Undefined Behavior: Data race detected between (1) non-atomic write on thread `unnamed-ID` and (2) non-atomic read on thread `unnamed-ID` at ALLOC+0x8. (2) just happened here
+   |
+   |
+LL |         *self.lock_count.get() = (*self.lock_count.get()).checked_add(1)?;
+   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^ Data race detected between (1) non-atomic write on thread `unnamed-ID` and (2) non-atomic read on thread `unnamed-ID` at ALLOC+0x8. (2) just happened here
+   |
+help: and (1) occurred earlier here
+   |
+LL |         println!("finished waiting for initialization");
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
+   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
+   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
+   = note: BACKTRACE (of the first span) on thread `unnamed-ID`:
+   = note: inside `std::sync::ReentrantLock::<std::cell::RefCell<std::io::LineWriter<std::io::stdio::StdoutRaw>>>::increment_lock_count` at RUSTLIB/std/src/sync/reentrant_lock.rs:LL:CC
+   = note: inside `std::sync::ReentrantLock::<std::cell::RefCell<std::io::LineWriter<std::io::stdio::StdoutRaw>>>::lock` at RUSTLIB/std/src/sync/reentrant_lock.rs:LL:CC
+   = note: inside `<&std::io::Stdout as std::io::Write>::write_fmt` at RUSTLIB/std/src/io/stdio.rs:LL:CC
+   = note: inside `<std::io::Stdout as std::io::Write>::write_fmt` at RUSTLIB/std/src/io/stdio.rs:LL:CC
+   = note: inside `std::io::stdio::print_to::<std::io::Stdout>` at RUSTLIB/std/src/io/stdio.rs:LL:CC
+   = note: inside `std::io::_print` at RUSTLIB/std/src/io/stdio.rs:LL:CC
---
+error: aborting due to 1 previous error
+


error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass-dep\concurrency\windows_init_once.stdout` to the actual output
+++ <stdout output>
 completing initialization
 finished waiting for initialization
-finished waiting for initialization
-finished waiting for initialization
-failing initialization
-retrying initialization
-finished waiting for initialization


error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
    Error: Undefined Behavior: Data race detected between (1) non-atomic write on thread `unnamed-1` and (2) non-atomic read on thread `unnamed-2` at alloc5320+0x8. (2) just happened here
full stderr:
full stderr:
error: Undefined Behavior: Data race detected between (1) non-atomic write on thread `unnamed-1` and (2) non-atomic read on thread `unnamed-2` at alloc5320+0x8. (2) just happened here
   |
   |
LL |         *self.lock_count.get() = (*self.lock_count.get()).checked_add(1)?;
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^ Data race detected between (1) non-atomic write on thread `unnamed-1` and (2) non-atomic read on thread `unnamed-2` at alloc5320+0x8. (2) just happened here
   |
help: and (1) occurred earlier here
   |
LL |         println!("finished waiting for initialization");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE (of the first span) on thread `unnamed-2`:
   = note: inside `std::sync::ReentrantLock::<std::cell::RefCell<std::io::LineWriter<std::io::stdio::StdoutRaw>>>::increment_lock_count` at C:\a\rust\rust\library\std\src\sync\reentrant_lock.rs:244:34: 244:58
   = note: inside `std::sync::ReentrantLock::<std::cell::RefCell<std::io::LineWriter<std::io::stdio::StdoutRaw>>>::lock` at C:\a\rust\rust\library\std\src\sync\reentrant_lock.rs:188:17: 188:44
   = note: inside `<&std::io::Stdout as std::io::Write>::write_fmt` at C:\a\rust\rust\library\std\src\io\stdio.rs:776:9: 776:20
   = note: inside `<std::io::Stdout as std::io::Write>::write_fmt` at C:\a\rust\rust\library\std\src\io\stdio.rs:750:9: 750:33
   = note: inside `std::io::stdio::print_to::<std::io::Stdout>` at C:\a\rust\rust\library\std\src\io\stdio.rs:1087:21: 1087:47
   = note: inside `std::io::_print` at C:\a\rust\rust\library\std\src\io\stdio.rs:1164:5: 1164:37
```

</details>

--- a/library/std/src/thread/mod.rs
+++ b/library/std/src/thread/mod.rs
@@ -690,7 +690,7 @@ pub(crate) fn set_current(thread: Thread) {
     // control over where this is called, so just abort if there is a bug.
     CURRENT.with(|current| match current.set(thread) {
         Ok(()) => {}
-        Err(_) => rtabort!("should only be set once"),
+        Err(_) => rtabort!("thread::set_current should only be called once per thread"),
     });
 }
 

